core/editor/src/document/document_message_handler.rs

```diff
@@ -100,29 +100,91 @@ impl MessageHandler<DocumentMessage, ()> for DocumentMessageHandler {
 			}
 			CloseDocument(id) => {
 				assert!(id < self.documents.len(), "Tried to select a document that was not initialized");
-				self.active_document = id - 1;
-				self.documents.pop();
-				let lp = self.active_document_mut().layer_panel(&[]).expect("Could not get panel for active doc");
-				responses.push_back(FrontendMessage::ExpandFolder { path: Vec::new(), children: lp }.into());
-				responses.push_back(FrontendMessage::SetActiveDocument { document_index: self.active_document }.into());
+				// Remove doc from the backend store. Use 'id' as FE tabs and BE documents will be in sync.
+				self.documents.remove(id);
 				responses.push_back(FrontendMessage::CloseDocument { document_index: id }.into());
-				responses.push_back(
-					FrontendMessage::UpdateCanvas {
-						document: self.active_document_mut().document.render_root(),
+
+				if self.documents.len() > 0 {
+					// If we have atleat one doc remaining, just select the correct doc.
```

Comment:
atleat -> atleast (also we usually don't end comments with a period but that's an unimportant nitpick)

I will review the rest of this later for you but I wanted to point out this typo in case I missed it on a later pass.